File: backend/models/eventModel.js
// models/eventModel.js
const mongoose = require('mongoose');

const rewardSchema = new mongoose.Schema({
    points: { type: Number, default: 0 },
    items: [{
        itemId: { type: mongoose.Schema.Types.ObjectId, ref: 'Item' },
        quantity: { type: Number, default: 1 }
    }],
    title: { type: String }
}, { _id: false });

const optionSchema = new mongoose.Schema({
    text: { type: String, required: true },
    rewards: { type: rewardSchema, required: true },
    consume: { type: rewardSchema }
}, { _id: false });

const eventSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true,
    },
    description: {
        type: String,
        required: true,
    },
    type: {
        type: String,
        enum: ['daily', 'puzzle', 'game', 'chest', 'bless'],
        required: true,
    },
    spotId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Spot',
        required: false,
    },
    options: [optionSchema],
    rewards: rewardSchema,
    consume: rewardSchema
}, { timestamps: true });

const Event = mongoose.model('Event', eventSchema);
module.exports = Event;

File: frontend/app/src/main/java/com/ntou01157/hunter/models/model_api/EventModel.kt
// frontend/app/src/main/java/com/ntou01157/hunter/models/model_api/EventModel.kt
package com.ntou01157.hunter.models.model_api

import com.google.gson.annotations.SerializedName

data class EventModel(
    @SerializedName("_id")
    val id: String,
    val name: String,
    val description: String,
    val type: String,
    val spotId: String? = null,
    val options: List<Option>? = null,
    val rewards: Reward? = null,
    val consume: Reward? = null,
    val lastTriggered: String? = null,
)

data class Option(
    val text: String,
    val rewards: Reward,
    val consume: Reward? = null
)

data class Reward(
    val points: Int? = 0,
    val items: List<RewardItemDetail>? = null,
    val title: String? = null
)

data class RewardItemDetail(
    val itemId: String,
    val quantity: Int
)